src/types.rs

```diff
@@ -32,6 +34,38 @@ pub struct Amount {
     pub micro_ccd: u64,
 }
 
+#[cfg(all(feature = "std", target_arch = "wasm32"))]
+use std::sync::atomic::AtomicU8;
+#[cfg(all(feature = "std", target_arch = "wasm32"))]
+static ADDRESS_COUNTER: AtomicU8 = AtomicU8::new(0);
+#[cfg(all(feature = "std", target_arch = "wasm32"))]
+use getrandom::register_custom_getrandom;
+#[cfg(all(feature = "std", target_arch = "wasm32"))]
+fn custom_random(dest: &mut [u8]) -> Result<(), getrandom::Error> {
+    for byte in dest.iter_mut() {
+        *byte = ADDRESS_COUNTER.load(std::sync::atomic::Ordering::SeqCst);
+        ADDRESS_COUNTER.fetch_add(1, std::sync::atomic::Ordering::SeqCst);
+    }
+    Ok(())
+}
+#[cfg(all(feature = "std", target_arch = "wasm32"))]
+register_custom_getrandom!(custom_random);
```

Comment:
Could we instead make use of special host functions for this?

Host functions that are only available during testing with cargo-concordium-test. The same way that we currently do with `report_error`. The idea being that `cargo-concordium` selects the seed and then generates random values from it and we register that host function with quickcheck here.

Would that work? We can discuss the details if my message above is cryptic.